features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/ecp/ecp_internal_alt.c

```diff
@@ -0,0 +1,828 @@
+     * 0 <= X < P
+     * 0 <= y < P 
+     */
+    if (mbedtls_mpi_cmp_int(&pt->Z, 1) == 0 &&
+        mbedtls_mpi_cmp_int(&pt->X, 0) >= 0 &&
+        mbedtls_mpi_cmp_mpi(&pt->X, &grp->P) < 0 &&
+        mbedtls_mpi_cmp_int(&pt->Y, 0) >= 0 &&
+        mbedtls_mpi_cmp_mpi(&pt->Y, &grp->P) < 0) {
+        return 0;
+    }
+
+    int ret;
+    mbedtls_mpi N, Zi, ZZi;
+    const mbedtls_mpi *Np;
+
+    mbedtls_mpi_init(&N);
+    mbedtls_mpi_init(&Zi);
+    mbedtls_mpi_init(&ZZi);
+
+    /* Use MBEDTLS_INTERNAL_MPI_NORM(Np, N1, N_, P) to get normalized MPI
+     *
+     * N_: Holds normalized MPI if the passed-in MPI N1 is not
+     * Np: Pointer to normalized MPI, which could be N1 or N_
+     */
+
+    /* Zi = 1 / Z */
+    mbedtls_mpi_lset(&Zi, 1);
+    MBEDTLS_INTERNAL_MPI_NORM(&Np, pt->Z, N, grp->P);
+    MBEDTLS_MPI_CHK(mbedtls_internal_run_modop(&Zi, &Zi, Np, &grp->P, grp->pbits, MODOP_DIV));
+    
+    /* ZZi = 1 / Z^2 = Zi * Zi */
+    MBEDTLS_MPI_CHK(mbedtls_internal_run_modop(&ZZi, &Zi, &Zi, &grp->P, grp->pbits, MODOP_MUL));
+    
+    /* X = X / Z^2 = X * ZZi */
+    MBEDTLS_INTERNAL_MPI_NORM(&Np, pt->X, N, grp->P);
+    MBEDTLS_MPI_CHK(mbedtls_internal_run_modop(&pt->X, Np, &ZZi, &grp->P, grp->pbits, MODOP_MUL));
+    
+    /* Y = Y / Z^3 = Y * ZZi * Zi */
+    MBEDTLS_INTERNAL_MPI_NORM(&Np, pt->Y, N, grp->P);
+    MBEDTLS_MPI_CHK(mbedtls_internal_run_modop(&pt->Y, Np, &ZZi, &grp->P, grp->pbits, MODOP_MUL));
+    MBEDTLS_MPI_CHK(mbedtls_internal_run_modop(&pt->Y, &pt->Y, &Zi, &grp->P, grp->pbits, MODOP_MUL));
+    
+    /* Z = 1 */
+    MBEDTLS_MPI_CHK(mbedtls_mpi_lset(&pt->Z, 1));
+
+cleanup:
+
+    mbedtls_mpi_free(&ZZi);
+    mbedtls_mpi_free(&Zi);
+    mbedtls_mpi_free(&N);
+
+    return ret;
+}
+#endif
+
+#if defined(MBEDTLS_ECP_NORMALIZE_JAC_MANY_ALT)
+/**
+ * \brief           Normalize jacobian coordinates of an array of (pointers to)
+ *                  points.
+ *
+ * \param grp       Pointer to the group representing the curve.
+ *
+ * \param T         Array of pointers to the points to normalise.
+ *
+ * \param t_len     Number of elements in the array.
+ *
+ * \return          0 if successful.
+ */
+int mbedtls_internal_ecp_normalize_jac_many(const mbedtls_ecp_group *grp,
+                                            mbedtls_ecp_point *T[], size_t t_len)
+{
+    if (T == NULL || t_len == 0) {
+        return MBEDTLS_ERR_ECP_BAD_INPUT_DATA;
+    }
+    
+    int ret;
+    
+    mbedtls_ecp_point **ecp_point = T;
+    mbedtls_ecp_point **ecp_point_end = T + t_len;
+    
+    for (; ecp_point != ecp_point_end; ecp_point ++) {
+        MBEDTLS_MPI_CHK(mbedtls_internal_ecp_normalize_jac(grp, *ecp_point));
+    }
+
+cleanup:
+
+    return ret;
+}
+#endif
+
+int mbedtls_internal_run_eccop(const mbedtls_ecp_group *grp,
+                            mbedtls_ecp_point *R,
+                            const mbedtls_mpi *m,
+                            const mbedtls_ecp_point *P,
+                            const mbedtls_mpi *n,
```

Comment:
The parameter `n` is unused.

Reply:
@hanno-arm The parameter `n` is kept to be consistent with `R = m*P + n*Q`. I add comment for it in https://github.com/ARMmbed/mbed-os/pull/5812/commits/17280372a7200b407615cd279f6cf0ea1992f1b8.